python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -0,0 +1,124 @@
+    MapOperatorTasksImpl,
+)
+from ray.data._internal.execution.operators.map_operator_actors_impl import (
+    MapOperatorActorsImpl,
+)
+
+
+class MapOperator(PhysicalOperator):
+    """A streaming operator that maps input bundles 1:1 to output bundles.
+
+    This operator implements the distributed map operation, supporting both task
+    and actor compute strategies.
+    """
+
+    def __init__(
+        self,
+        transform_fn: Callable[[Iterator[Block]], Iterator[Block]],
+        input_op: PhysicalOperator,
+        name: str = "Map",
+        compute_strategy: Optional[ComputeStrategy] = None,
+        min_rows_per_batch: Optional[int] = None,
+        ray_remote_args: Optional[Dict[str, Any]] = None,
+    ):
+        """Create a MapOperator.
+
+        Args:
+            transform_fn: The function to apply to each ref bundle input.
+            input_op: Operator generating input data for this op.
+            name: The name of this operator.
+            compute_strategy: Customize the compute strategy for this op.
+            min_rows_per_batch: The number of rows to gather per batch passed to the
+                transform_fn, or None to use the block size. Setting the batch size is
+                important for the performance of GPU-accelerated transform functions.
+            ray_remote_args: Customize the ray remote args for this op's tasks.
+        """
+        self._transform_fn = transform_fn
+        self._strategy = compute_strategy or TaskPoolStrategy()
+        self._remote_args = (ray_remote_args or {}).copy()
+        self._output_metadata: List[BlockMetadata] = []
+        self._min_rows_per_batch = min_rows_per_batch
+        if isinstance(self._strategy, TaskPoolStrategy):
+            self._execution_state = MapOperatorTasksImpl(self)
+        elif isinstance(self._strategy, ActorPoolStrategy):
+            self._execution_state = MapOperatorActorsImpl(self)
+        else:
+            raise NotImplementedError
+        super().__init__(name, [input_op])
+
+    def get_transform_fn(
+        self,
+    ) -> Callable[[Iterator[Block]], Iterator[Block]]:
+        """Return the block transformation to run on a worker process.
+
+        This callable must be serializable as it will be sent to remote processes.
+
+        Returns:
+            A callable taking an iterator over input blocks of a RefBundle. Typically,
+            this will yield only a single block, unless the transformation has
+            multiple inputs. It is an iterator for memory efficiency.
+        """
+        return self._transform_fn
+
+    # TODO(ekl): slim down ComputeStrategy to only specify the compute
+    # config and not contain implementation code.
+    def compute_strategy(self) -> ComputeStrategy:
+        """Return the compute strategy to use for executing these tasks.
+
+        Supported strategies: {TaskPoolStrategy, ActorPoolStrategy}.
+        """
+        return self._strategy
```

Comment:
Hmm I wonder if we can keep the implementation details of compute strategy and ray remote args etc outside of the operators? It could be cleaner if we pass in the ray.remote Callable instead of the worker's Callable as the `transform_fn` but not sure if this will work so I'll leave it up to you.

Reply:
Yeah, I have a TODO on line 78 to clean this up in the future. I'm hoping the ComputeStrategy can turn into a simple dataclass once we migrate fully to the new backend. Right now, I avoided doing this refactoring to keep the changes self-contained.

About the callable, I think that's possible but it's probably also easier to do once we have the logical optimization layer in place (the optimizer could generate the ray.remote callable).